binfmt/libelf/libelf_coredump.c

```diff
@@ -280,9 +279,10 @@ static void elf_emit_tcb_note(FAR struct elf_dumpinfo_s *cinfo,
       regs = (uintptr_t *)tcb->xcp.regs;
     }
 
+#if defined(ARCH_ARM) || defined(ARCH_ARM64)
   if (regs != NULL)
     {
-      for (i = 0; i < nitems(status.pr_regs); i++)
+      for (int i = 0; i < nitems(status.pr_regs); i++)
```

Comment:
generic code on NuttX should follow C89 because old compilers don't support C99 or newer